test/acceptance/framework/consul_cluster.go

```diff
@@ -144,29 +143,25 @@ func (h *HelmCluster) SetupConsulClient(t *testing.T, secure bool) *api.Client {
 	remotePort := 8500 // use non-secure by default
 
 	if secure {
-		// overwrite remote port to HTTPS
+		// Overwrite remote port to HTTPS.
 		remotePort = 8501
 
-		// get the CA
-		caSecret, err := h.kubernetesClient.CoreV1().Secrets(namespace).Get(h.releaseName+"-consul-ca-cert", metav1.GetOptions{})
-		require.NoError(t, err)
-		caFile, err := ioutil.TempFile("", "")
-		require.NoError(t, err)
-		helpers.Cleanup(t, h.noCleanupOnFailure, func() {
-			require.NoError(t, os.Remove(caFile.Name()))
-		})
-
-		if caContents, ok := caSecret.Data["tls.crt"]; ok {
-			_, err := caFile.Write(caContents)
+		// Get the ACL token. First, attempt to read it from the bootstrap token.
+		// If the bootstrap token doesn't exist, try to read the replication token from the federation secret.
+		aclSecret, err := h.kubernetesClient.CoreV1().Secrets(namespace).Get(h.releaseName+"-consul-bootstrap-acl-token", metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			federationSecret := fmt.Sprintf("%s-consul-federation", h.releaseName)
+			aclSecret, err = h.kubernetesClient.CoreV1().Secrets(namespace).Get(federationSecret, metav1.GetOptions{})
+			require.NoError(t, err)
+			config.Token = string(aclSecret.Data["replicationToken"])
+		} else if err == nil {
+			config.Token = string(aclSecret.Data["token"])
+		} else {
 			require.NoError(t, err)
 		}
 
-		// get the ACL token
-		aclSecret, err := h.kubernetesClient.CoreV1().Secrets(namespace).Get(h.releaseName+"-consul-bootstrap-acl-token", metav1.GetOptions{})
-		require.NoError(t, err)
-
-		config.TLSConfig.CAFile = caFile.Name()
-		config.Token = string(aclSecret.Data["token"])
+		// It's OK to skip TLS verification for local traffic.
+		config.TLSConfig.InsecureSkipVerify = true
 		config.Scheme = "https"
```

Comment:
These lines make more sense to me to be grouped with `remotePort = 8501`

Reply:
I've moved that whole section to be together with remotePort.